### ISSUE TITLE
SourceRoots not set correctly from NuGetPackageFolders

### ISSUE THREAD

**Issue by tmat** [NuGet/Home#9810]
The generated props file for the project (`git-test1\obj\git-test1.csproj.nuget.g.props`) contains:

```xml
    <NuGetPackageFolders Condition=" '$(NuGetPackageFolders)' == '' ">C:\Users\<user-name>\.nuget\packages\;C:\Program Files\dotnet\sdk\NuGetFallbackFolder</NuGetPackageFolders>
  </PropertyGroup>
...
  <ItemGroup Condition=" '$(ExcludeRestorePackageImports)' != 'true' ">
    <SourceRoot Include="$([MSBuild]::EnsureTrailingSlash($(NuGetPackageFolders)))" />
  </ItemGroup>
```

This adds a **single** item to SourceRoot with ItemSpec = `C:\Users\Tomas\.nuget\packages\;C:\Program Files\dotnet\sdk\NuGetFallbackFolder\`. Instead, we need 2 items: `C:\Users\<user-name>\.nuget\packages\` and `C:\Program Files\dotnet\sdk\NuGetFallbackFolder\`.

Possible fix:
```xml
  <ItemGroup Condition=" '$(ExcludeRestorePackageImports)' != 'true' ">
     <_NuGetPackageFolders Include="$(NuGetPackageFolders)" NormalizedPath="$([MSBuild]::EnsureTrailingSlash(%(Identity)))" />
    <SourceRoot Include="@(_NuGetPackageFolders->'%(NormalizedPath)')" />
  </ItemGroup>
```

**Comment by tmat** [comment]
While testing end-to-end scenarios I also discovered issue https://github.com/dotnet/roslyn/issues/46061.
So it's actually good this does not have an effect at this point since it would override Source Link entries for source packages that set them. Fixing both issues in 16.8 seems best.

**Comment by nkolev92** [comment]
Per internal thread, wait on dotnet/roslyn#46061 to be fixed before fixing this. 
Adding a blockedbyexternal label + added it to Sprint 174.

**Comment by nkolev92** [comment]
bumping to S176 as the roslyn side is not scheduled yet.

**Comment by nkolev92** [comment]
Bumping S177, as the roslyn side is not yet ready :)

**Comment by nkolev92** [comment]
S179...blocking issue not fixed yet.

**Comment by nkolev92** [comment]
Adding it to 181 since the roslyn side fix is out.

**Comment by nkolev92** [comment]
Fix in PR https://github.com/NuGet/NuGet.Client/pull/3815

**Comment by bording** [comment]
@nkolev92 Has this fix shipped yet? If so, what version includes it?

As some general feedback, it's really difficult to tell when changes get released because nothing in your issues labels or milestones make that clear.

**Comment by nkolev92** [comment]
It was fixed in 5.9, https://docs.microsoft.com/en-us/nuget/release-notes/nuget-5.9. 

You need zenhub to view the list of bugs fixed. Common feedback for us. 

cc @zkat @aortiz-msft
